src/main/java/com/nirmal/banking/service/UserService.java

```diff
@@ -84,8 +85,70 @@ public String uploadImage(MultipartFile[] file, HttpServletRequest request) thro
         return SuccessMessages.UPLOADED;
     }
 
-    void convertMultipartFileToFile(MultipartFile[] multipartFiles, String filePath) throws IOException {
+    public String depositAmount(HttpServletRequest request, Integer depositAmount) {
+        String uid = extractUid(request);
+        CustomUserDetails customUserDetails = userDetailsRepo.findByuid(uid);
 
+        if (!customUserDetails.getKycStatus().equals(KycStatus.APPROVED))
+            throw new CustomException(ErrorMessages.KYC_NOT_APPROVED);
+
+        if (!transactionRepo.existsByUid(uid))
+            if (depositAmount < 1000) throw new CustomException(ErrorMessages.MINIMUM_DEPOSIT_AMOUNT);
+
+        TransactionDetails transactionDetails = new TransactionDetails();
+        TransactionDetailsDto transactionDetailsDto = TransactionDetailsDto.builder()
```

Comment:
why do we need DTO here?

Reply:
we are inserting data into  table , through dto we done the process.